### PR TITLE
v2.2.0 Fix enum serialization and model paths for pypi

### PR DESCRIPTION
**Pull request by conorbergman**
### v2.2.0 Changes

- Add `__init__.py` to test and MergePythonSDK packages to fix pypi path traversal
- Add `none_type` as a valid type for all Optional fields
- Add `MergeEnumType` class all Model enums derive from
- Add conversion between `str` types and classes that derive from `MergeEnumType`

### CODE REVIEW

File: MergePythonSDK/accounting/model/account.py [hunk 1]
```diff
@@ -101,18 +101,19 @@ def openapi_types():
...
+            'type': (str, none_type, none_type,),  # noqa: E501
             'status': (AccountStatusEnum, str, none_type,),
-            'current_balance': (float, none_type,),  # noqa: E501
+            'current_balance': (float, none_type, none_type,),  # noqa: E501
             'currency': (CurrencyEnum, str, none_type,),
-            'account_number': (str, none_type,),  # noqa: E501
-            'remote_was_deleted': (bool,),  # noqa: E501
+            'account_number': (str, none_type, none_type,),  # noqa: E501
+            'parent_account': (str, none_type, none_type,),  # noqa: E501
+            'remote_was_deleted': (bool, none_type,),  # noqa: E501
```
Comment (conorbergman) [inline MergePythonSDK/accounting/model/account.py:116]: Add `none_type` to all optional fields

File: MergePythonSDK/accounting/model/account_details.py [hunk 2]
```diff
@@ -39,7 +39,7 @@
 
 
 def lazy_import():
-    from MergePythonSDK.accounting.model.category_enum import CategoryEnum
+    from MergePythonSDK.shared.model.category_enum import CategoryEnum
```
Comment (conorbergman) [inline MergePythonSDK/accounting/model/account_details.py:42]: move `CategoryEnum` to shared

File: MergePythonSDK/accounting/model/account_details_and_actions_status_enum.py [hunk 3]
```diff
@@ -36,9 +36,10 @@
 )
 from MergePythonSDK.shared.exceptions import ApiAttributeError
 from MergePythonSDK.shared.model_utils import import_model_by_name
+from MergePythonSDK.shared.model_utils import MergeEnumType
 
 
-class AccountDetailsAndActionsStatusEnum(ModelNormal):
+class AccountDetailsAndActionsStatusEnum(ModelNormal, MergeEnumType):
```
Comment (conorbergman) [inline MergePythonSDK/accounting/model/account_details_and_actions_status_enum.py:42]: Enum classes now subclass `MergeEnumType`

File: MergePythonSDK/accounting/model/account_details_and_actions_status_enum.py [hunk 4]
```diff
@@ -108,7 +109,7 @@ def discriminator():
 
     @classmethod
     @convert_js_args_to_python_args
-    def _from_openapi_data(cls, *args, **kwargs):  # noqa: E501
+    def _from_openapi_data(cls, value, *args, **kwargs):  # noqa: E501
```
Comment (conorbergman) [inline MergePythonSDK/accounting/model/account_details_and_actions_status_enum.py:112]: Add `value` as positional argument for Enums

File: test/basic_test.py [hunk 5]
```diff
@@ -127,6 +129,16 @@ def test_api_clients(self):
...
                 raise e
 
+        # Test ENUM serialization
+        ticketing_category = CategoriesEnum("ticketing")
+        assert ticketing_category.value == "ticketing"
+
+        # List LinkedAccounts test enum serialization and deserialization
+        api_client = ApiClient(ticketing_configuration)
+        linked_account_api_instance = LinkedAccountsApi(api_client)
+        api_response = linked_account_api_instance.linked_accounts_list(category=ticketing_category.value).results
```
Comment (conorbergman) [inline test/basic_test.py:141]: Test passing in CategoriesEnum

File: test/basic_test.py [hunk 6]
```diff
@@ -127,6 +129,16 @@ def test_api_clients(self):
...
+        assert api_response[0].category == CategoriesEnum("ticketing").value
```
Comment (conorbergman) [inline test/basic_test.py:142]: Test retrieving `CategoriesEnum` from response

File: MergePythonSDK/shared/model_utils.py [hunk 7]
```diff
@@ -1426,6 +1430,10 @@ def validate_and_convert_types(input_value, required_types_mixed, path_to_item,
     input_class_simple = get_simple_class(input_value)
     valid_type = is_valid_type(input_class_simple, valid_classes)
     if not valid_type:
+        if isinstance(input_value, str):
+            for _typ in valid_classes:
+                if issubclass(_typ, MergeEnumType):
+                    return _typ(input_value)
```
Comment (conorbergman) [inline MergePythonSDK/shared/model_utils.py]: Allow for type conversion from `str` to classes that subclass `MergeEnumType`

File: MergePythonSDK/accounting/model/account.py [hunk 8]
```diff
@@ -101,18 +101,19 @@ def openapi_types():
...
-            'id': (str,),  # noqa: E501
-            'remote_id': (str, none_type,),  # noqa: E501
-            'remote_data': ([RemoteData], none_type,),  # noqa: E501
-            'name': (str, none_type,),  # noqa: E501
-            'description': (str, none_type,),  # noqa: E501
+            'id': (str, none_type,),  # noqa: E501
+            'remote_id': (str, none_type, none_type,),  # noqa: E501
+            'remote_data': ([RemoteData], none_type, none_type,),  # noqa: E501
+            'name': (str, none_type, none_type,),  # noqa: E501
+            'description': (str, none_type, none_type,),  # noqa: E501
```
Comment (leewang0) [inline MergePythonSDK/accounting/model/account.py:108]: It's not great that `none_type` appears twice.

File: MergePythonSDK/accounting/model/category_enum.py [hunk 9]
```diff
@@ -36,9 +36,10 @@
 )
 from MergePythonSDK.shared.exceptions import ApiAttributeError
 from MergePythonSDK.shared.model_utils import import_model_by_name
+from MergePythonSDK.shared.model_utils import MergeEnumType
 
 
-class CategoryEnum(ModelNormal):
+class CategoryEnum(ModelNormal, MergeEnumType):
```
Comment (leewang0) [inline MergePythonSDK/accounting/model/category_enum.py:42]: I thought this class was moved to shared in the pr description?

File: test/basic_test.py (hunk 6 above)
Comment (leewang0) [reply test/basic_test.py:142]: For a non-category enum, can you add a test for deserializing a non-standard value? Like, we wont' return anything different here for a long while outside of the 5 standard categories, but for ats url.type or hris martial_status we sometimes return a value outside of the common ones. Just want to make sure that still works.

**Review by leewang0 — APPROVED** [review]

**Pull request merged**